app-i18n/librime/librime-1.9.0.ebuild

```diff
@@ -0,0 +1,52 @@
+# Copyright 2012-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="8"
+
+inherit cmake
+
+DESCRIPTION="RIME (Rime Input Method Engine) core library"
+HOMEPAGE="https://rime.im/ https://github.com/rime/librime"
+SRC_URI="https://github.com/rime/librime/archive/refs/tags/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0/1-${PV}"
+KEYWORDS="~amd64 ~x86"
+IUSE="debug test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	dev-cpp/glog:=
+	>=dev-libs/boost-1.74:=
+	app-i18n/opencc:0=
+	dev-cpp/yaml-cpp:0=
+	dev-libs/leveldb:0=
+	dev-libs/marisa:0=
+"
+DEPEND="${RDEPEND}
+	dev-libs/darts
+	dev-libs/utfcpp
+	test? ( dev-cpp/gtest )
+"
+BDEPEND=""
+
+DOCS=(CHANGELOG.md README.md)
+
+src_configure() {
+	local -x CXXFLAGS="${CXXFLAGS} -I${ESYSROOT}/usr/include/utf8cpp"
+
+	# for glog
+	if use debug; then
+		CXXFLAGS+=" -DDCHECK_ALWAYS_ON"
+	else
+		CXXFLAGS+=" -DNDEBUG"
+	fi
```

Comment:
Can't `CMAKE_BUILD_TYPE` do the job?

Reply:
Not sure if CMAKE_BUILD_TYPE can achieve the same situation. These are defined in the glog library.

Reply:
`-DDCHECK_ALWAYS_ON` is for glog right? But the librime is linked against system `dev-cpp/glog`.
Upstream git submodule is not include in src tarball. Is setting this flag do anything useful?

Reply:
If I understand correctly, it includes Glog's header file (logging.h) at compile time, uses the DLOG macro, so it's valid.

Reply:
Still, I think `CMAKE_BUILD_TYPE=debug` should be set for `debug` & append-cxxflags, and do nothing for normal build.

Reply:
Fixed, Thanks!

Reply:
https://github.com/gentoo/gentoo/blob/ede01cc9457106602b6f48aa9e76d06c07dd512e/eclass/cmake.eclass#L47-L58
`CMAKE_BUILD_TYPE` is set in cmake.eclass, `CMAKE_BUILD_TYPE=Gentoo` is only for EAPI=7
Something like `use debug && CMAKE_BUILD_TYPE=Debug` is better.

Reply:
You are right, fixed.